### PR TITLE
Use existing installation location if one exists

### PR DESCRIPTION
**Pull request by watsonian**
This adds an additional check to see if the Doppler CLI already exists on the PATH somewhere. If it does, it uses that as the installation location. It addresses problems introduced by Homebrew switching to installing under `/opt/homebrew` and will also make it so users who installed to a custom location using `--install-path` get updated properly when `doppler update` is run.

Closes ENG-2166

### CODE REVIEW

File: scripts/install.sh
Comment (Piccirello) [inline scripts/install.sh]: `which` isn't available on some platforms, so you'll want to use `command -v` here

File: scripts/install.sh
Comment (Piccirello) [inline scripts/install.sh]: Currently, if the CLI already exists but we're not able to overwrite it, we'll keep executing and ultimately install a second copy in one of the default paths. This is probably ok, but wanted to call it out.

**Review by Piccirello — COMMENTED** [review]
Can you add a test for this so we can ensure it doesn't break in the future? I'm thinking probably a test to ensure that the new logic works as expected when the CLI is in a non-default path, and maybe another test to ensure we continue execution when the existing binary can't be updated.

File: scripts/install.sh
Comment (watsonian) [reply scripts/install.sh]: Per our discussion we're going to leave this as-is. This situation shouldn't come up too frequently and if it does, it's not a huge issue. If it comes up as a problem later, we can revisit it.

**Comment by watsonian** [comment]
I think this is good to go now. I squished the commits down and have tests. Let me know if you see anything that could use some refinement.

**Review by Piccirello — DISMISSED** [review]
These tests look great. Nice job!

**Comment by rgmvisser** [comment]
Good stuff @watsonian ! :)

**Comment by Piccirello** [comment]
If you rebase onto `master` the Salus check should start passing

**Comment by watsonian** [comment]
Nice! The rebase fixed that. Everything looks good, but it needs one last code owner review before it'll let me merge.

**Review by Piccirello — APPROVED** [review]

**Pull request merged**
